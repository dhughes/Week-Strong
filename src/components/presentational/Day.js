import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import theme from '../../util/theme';

/*
  clock-start = day the fitness test is taken



 */

const DayBox = styled.div`
  component: DayBox;
  text-align: center;
  font-size: 1.2rem;
  width: 2.5rem;
  border-radius: 0.3em;

  color: ${theme.primaryText};
  border: 2px solid ${theme.primaryText};
  background-color: ${theme.background};

  line-height: 2.5rem;
  height: 2.5rem;
  padding: 0;
  margin: 0.25rem 0.25rem 0.25rem 0.25rem;

  &.notInProgram,
  &.afterToday {
    opacity: 0.25;
    border-width: 1px;
  }

  &.workoutDay {
    background-color: ${theme.positive.lighten(1.25)};
  }

  &.workedOut {
    background-color: ${theme.positive};
    color: ${theme.primaryText.negate()};
  }

  &.fitnessTestDay {
    opacity: 1;
  }

  &.today {
    font-weight: bold;
    border: 2px solid ${theme.primaryText};
    box-shadow: 0px 0px 0px 2px ${theme.primaryText};
    z-index: 1;
  }

  &.missed {
    background-color: ${theme.negative.lighten(1)};
  }

  &.disabled {
    border: 1px solid ${theme.primaryText.fade(0.75)};
  }

  &.beforeStart {
    border-width: 2px;
    border-style: dashed;
  }

  &.makeupDay {
    background-color: ${theme.positive}!important;
  }
`;

const days = ['SUNDAY', 'MONDAY', 'TUESDAY', 'WEDNESDAY', 'THURSDAY', 'FRIDAY', 'SATURDAY'];
export const months = [
  'January',
  'Febuary',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December'
];

const Day = props =>
  <DayBox
    className={`
      ${!props.inProgram ? 'notInProgram' : ''}
      ${props.isWorkoutDay ? 'workoutDay' : ''}
      ${props.isMakeupDay ? 'makeupDay' : ''}
      ${props.workedOut ? 'workedOut' : ''}
      ${props.missed ? 'missed' : ''}
      ${props.isToday ? 'today' : ''}
      ${props.isBeforeStart ? 'beforeStart' : ''}
      ${props.isFitnessTestDay ? 'fitnessTestDay' : ''}
      ${props.isAfterToday ? 'afterToday' : ''}
      ${props.disabled ? 'disabled' : ''}`}
    data-value={days.indexOf(props.day)}
    title={props.day}
    {...props}
  >
    {props.day.substr(0, 1)}
  </DayBox>;

Day.defaultProps = {
  onClick: e => {},
  disabled: false
};

Day.propTypes = {
  inProgram: PropTypes.bool,
  isWorkoutDay: PropTypes.bool,
  isMakeupDay: PropTypes.bool,
  workedOut: PropTypes.bool,
  missed: PropTypes.bool,
  isToday: PropTypes.bool,
  day: PropTypes.oneOf(days),
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  isBeforeStart: PropTypes.bool,
  isFitnessTestDay: PropTypes.bool,
  isAfterToday: PropTypes.bool
};

export default Day;
