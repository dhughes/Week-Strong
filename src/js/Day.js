import React from 'react';
import styled, { css } from 'styled-components';

const Container = styled.button`
  flex: 1;
  text-align: center;
  box-sizing: border-box;
  color: #EEEBD3;
  font-size: 1em;

  ${props => props.isSelected && css`
      border: 4px solid #69995D;
      background-color: #69995D;
      color: #393E41;
  `};

  ${props => props.wasMissed && css`
      border: 4px solid #B02E0C;
      background-color: #B02E0C;
      color: #EEEBD3;
  `};

  ${props => props.isToday && css`
      border: 4px solid #EEEBD3;
      font-weight: bold;
  `};

`;

const Day = props => {
  return (
    <Container
      isToday={props.dayOfWeek === props.today}
      isSelected={props.isSelected}
      wasMissed={props.wasMissed}
      onClick={props.onClick}
      value={props.dayOfWeek}
    >
      {getDay(props.dayOfWeek).substring(0, 1).toUpperCase()}
    </Container>
  );
};

Day.defaultProps = {
  isSelected: false,
  wasMissed: false,
  onClick: () => {}
};

const getDay = day => {
  switch (day) {
    case 0:
      return 'Sunday';
    case 1:
      return 'Monday';
    case 2:
      return 'Tuesday';
    case 3:
      return 'Wednesday';
    case 4:
      return 'Thursday';
    case 5:
      return 'Friday';
    case 6:
      return 'Saturday';
    default:
      return '';
  }
};

export default Day;
