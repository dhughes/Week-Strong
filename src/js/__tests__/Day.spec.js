import React from 'react';
import { shallow, mount } from 'enzyme';
import Day from '../Day';

test('Day renders', () => {
  const component = shallow(<Day dayOfWeek={2} />);
  expect(component).toMatchSnapshot();
});

test('Day shows letter of day of week', () => {
  const component = mount(<Day dayOfWeek={1} />);
  expect(component.find('button').text()).toEqual('M');
});

test('Day shows uppercase letter of day of week', () => {
  const component = mount(<Day dayOfWeek={3} />);
  expect(component.find('button').text()).toEqual('W');
});
