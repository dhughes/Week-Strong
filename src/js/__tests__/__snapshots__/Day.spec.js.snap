// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day renders 1`] = `
<div>
  Day
</div>
`;
